internal/pkg/cli/pipeline_package_test.go

```diff
@@ -0,0 +1,326 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/aws/copilot-cli/internal/pkg/cli/mocks"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+type packagePipelineMocks struct {
+	store                  *mocks.Mockstore
+	prompt                 *mocks.Mockprompter
+	prog                   *mocks.Mockprogress
+	deployer               *mocks.MockpipelineDeployer
+	pipelineStackConfig    *mocks.MockpipelineStackConfig
+	ws                     *mocks.MockwsPipelineReader
+	actionCmd              *mocks.MockactionCommand
+	deployedPipelineLister *mocks.MockdeployedPipelineLister
+}
+
+func TestPipelinePackageOpts_Execute(t *testing.T) {
+	const (
+		appName         = "badgoose"
+		region          = "us-west-2"
+		accountID       = "123456789012"
+		pipelineName    = "pipepiper"
+		badPipelineName = "pipeline-badgoose-honkpipes"
+		relativePath    = "/copilot/pipelines/pipepiper/manifest.yml"
+	)
+	pipeline := workspace.PipelineManifest{
+		Name: pipelineName,
+		Path: "copilot/pipeline.yml",
+	}
+	mockPipelineManifest := &manifest.Pipeline{
+		Name:    "pipepiper",
+		Version: 1,
+		Source: &manifest.Source{
+			ProviderName: "GitHub",
+			Properties: map[string]interface{}{
+				"repository": "aws/somethingCool",
+				"branch":     "main",
+			},
+		},
+		Stages: []manifest.PipelineStage{
+			{
+				Name:         "chicken",
+				TestCommands: []string{"make test", "echo 'made test'"},
+			},
+			{
+				Name:         "wings",
+				TestCommands: []string{"echo 'bok bok bok'"},
+			},
+		},
+	}
+
+	app := config.Application{
+		AccountID: accountID,
+		Name:      appName,
+		Domain:    "amazon.com",
+	}
+
+	mockResources := []*stack.AppRegionalResources{
+		{
+			S3Bucket:  "someBucket",
+			KMSKeyARN: "someKey",
+		},
+	}
+	mockEnv := &config.Environment{
+		Name:      "test",
+		App:       appName,
+		Region:    region,
+		AccountID: accountID,
+	}
+
+	testCases := map[string]struct {
+		inApp          *config.Application
+		inAppName      string
+		inPipelineName string
+		inRegion       string
+		inPipelineFile string
```

Comment:
I think we can simplify the unit tests by removing _some_ (I didn't check every one of them) of these inputs because we don't test extra things by giving different inputs.

For example, for each test case, no matter what `inAppName` we give, the logic will be exactly the same. As a result, we can probably just set it [here](https://github.com/aws/copilot-cli/pull/5029/files#diff-1acf3bb0c37dfcaf7dcf95ea9f1c9d7388f2dfe917673c7ba5c9145a05025c88R291)

Reply:
Makes sense, I have updated the test cases.